crates/symbolicator-service/src/caching/fs.rs

```diff
@@ -201,6 +234,20 @@ impl Cache {
         })
     }
 
+    /// Reads [`Metadata`] for the cache entry in `path` from the file at `path.md`.
+    ///
+    /// If the metadata file doesn't exist, this returns `Ok(None)`.
+    pub fn read_metadata(&self, path: &Path) -> io::Result<Option<Metadata>> {
+        let md_path = metadata_path(path);
+
+        catch_not_found(|| {
+            let file = File::open(&md_path)?;
+            let reader = BufReader::new(file);
+            serde_json::from_reader(reader)
```

Comment:
Maybe (eventually): Just explore what other serialization formats we can use.

Reply:
MessagePack works well in my local tests, and results in much smaller files.